servant-server/src/Servant/Server.hs

```diff
@@ -126,8 +138,8 @@ import           Servant.Server.Internal
 -- > main :: IO ()
 -- > main = Network.Wai.Handler.Warp.run 8080 app
 --
-serve :: (HasServer api '[]) => Proxy api -> Server api -> Application
-serve p = serveWithContext p EmptyContext
+serve :: (HasServer api DefaultErrorFormatters) => Proxy api -> Server api -> Application
+serve p = serveWithContext p defaultErrorFormatters
```

Comment:
I think it might be a good idea to have `serveWithContext p foo == serveWithContextOld p (foo ++ defaultErrorFormatters)` (`++` is a not-yet-existent version of the list concatenation but for contexts; and serverWithContextOld is the serveWithContext that we currently have).

Or, in words: I think the `serveWithContext` we expose should cons whatever context it's given to the default context (which has error formatters). That way it stays backwards compatible (and will continue to if we keep doing things in this direction); but since it picks the first context, you can still override the behaviour.

Reply:
Nice idea! I guess depending on something like `vinyl` for this functionality would be too heavy for `servant`?

I'll roll my own `++` here.